File: 5_percent/02_even_fibonacci_numbers.py
def even_fibonacci_numbers(max):
    """
    max: max value that value list should obtain
    return: sum of even numbers of the fibonacci sequence below max
    """
    value = [1, 2]
    while value[-1] < max:
        value.append(value[-2]+value[-1])
    return sum([elt for elt in value if elt % 2 == 0])

assert even_fibonacci_numbers(4000000) == 4613732, "even_fibonacci_numbers(4000000), 4613732"

File: 5_percent_cpp/02_even_fibonacci_numbers.cpp
// Pretty messy for now and ressembles a lot the C version but
// I'm just learning and I hope to improve soon

#include <iostream>

using namespace std;

int main(){
  int sum, fiba, fibb, list_size, max;
  sum = 2;

  fiba = 1;
  fibb = 2;
  list_size = 2;

  max = 4000000;

  while (fiba < max && fibb < max){
    if (list_size % 2 == 0){
      fiba += fibb;
      if (fiba % 2 == 0)
        sum += fiba;
    }
    else{
      fibb += fiba;
      if (fibb % 2 == 0)
        sum += fibb;
    }
    ++list_size;
  }

  cout << "The sum of even fibonacci numbers under " << max << " is " << sum << endl;
}